xnat_tools/dicom_export.py

```diff
@@ -6,178 +6,108 @@
 
 from xnat_tools.bids_utils import (
     assign_bids_name,
     bidsmap_scans,
-    populate_bidsmap,
     prepare_bids_prefixes,
     prepare_export_output_path,
 )
-
-from xnat_tools.xnat_utils import (
-    filter_scans,
-    get_project_and_subject_id,
-    get_scan_ids,
-)
+from xnat_tools.logging import setup_logging
+from xnat_tools.xnat_utils import filter_scans, get_project_and_subject_id, get_scan_ids
 
 _logger = logging.getLogger(__name__)
-
-
-def parse_args(args):
-    """Parse command line parameters
-
-    Args:
-      args ([str]): command line parameters as list of strings
-
-    Returns:
-      :obj:`argparse.Namespace`: command line parameters namespace
-    """
-    parser = argparse.ArgumentParser(
-        description="Dump XNAT Session into a BIDS friendly directory"
-    )
-    parser.add_argument("--host", default="https://bnc.brown.edu/xnat", help="Host")
-    parser.add_argument("-u", "--user", help="XNAT username", required=True)
-    parser.add_argument(
-        "-p",
-        "--password",
-        type=XNATPass,
-        help="XNAT password",
-        default=XNATPass.DEFAULT,
-    )
-    parser.add_argument("--session", help="Session ID", required=True)
-    parser.add_argument(
-        "--session_suffix",
-        help="Suffix of the session for BIDS e.g, 01. This will produce a sesstion label of sess-01",
-        required=True,
-        type=str,
-    )
-    parser.add_argument(
-        "--bids_root_dir", help="Root output directory for BIDS files", required=True
-    )
-    parser.add_argument(
-        "--bidsmap_file",
-        help="Bidsmap JSON file to correct sequence names",
-        required=False,
-        default="",
-    )
-    parser.add_argument(
-        "--seqlist",
-        help="List of sequences from XNAT to run if don't want to process all seuqences",
-        required=False,
-        default=[],
-        nargs="*",  # 0 or more values expected => creates a list
-        type=int,
-    )
-    parser.add_argument(
-        "--skiplist",
-        help="List of sequences from XNAT to SKIP. Accepts a list --skiplist 1 2 3",
-        required=False,
-        default=[],
-        nargs="*",  # 0 or more values expected => creates a list
-        type=int,
-    )
-    parser.add_argument(
-        "--log_id",
+app = typer.Typer()
+
+
+@app.command()
+def dicom_export(
+    session: str = typer.Argument(
+        ..., help="XNAT Session ID, that is the Accession # for an experiment."
+    ),
+    bids_root_dir: str = typer.Argument(
+        ..., help="Root output directory for exporting the files"
+    ),
+    user: str = typer.Option(None, "-u", "--user", prompt=True, help="XNAT User"),
+    password: str = typer.Option(
+        None, "-p", "--pass", prompt=True, hide_input=True, help="XNAT Password"
+    ),
+    host: str = typer.Option(
+        "https://bnc.brown.edu/xnat", "-h", "--host", help="XNAT'sURL"
+    ),
+    session_suffix: str = typer.Option(
+        "01",
+        "-S",
```

Comment:
Ahh, I see you've changed `-S` from `-ss` here. Looks like there's at least one other script that needs the same change. That's why I was asking about a way to make all commands take the same arguments instead of copying and pasting them into new commands

Reply:
I totally agree with your point. I didn't see a straight forward way... we could discuss other possibilities